shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -930,6 +937,16 @@ - (UIInputViewController*)inputViewController {
   return _textInputPlugin.textInputDelegate;
 }
 
+- (BOOL)respondsToSelector:(SEL)selector {
+  if (@available(iOS 17.0, *)) {
+    // See the comment on this method.
+    if (selector == @selector(insertionPointColor)) {
+      return NO;
```

Comment:
We need to be careful if this breaks other functionalities that uses `insertionPointColor`.